File: crmuretek-frontend/src/types/Visit.ts
// src/types/Visit.ts
export interface Visit {
  id: number;
  consulta?: {
    id: number;
    problemDescription?: string;
    customer: {
      id: number;
      consulta?: {
        id: number;
        problemDescription?: string;
        requestDate: string;
      };
      name?: string;
      email?: string;
      phoneNumber?: string;
      address?: string;
      contactChannel?: string;
      source?: string;
      contactDate?: string;
      title?: string | null;
      observations?: string | null;
    };
    requestDate: string;
  };
  visitDate?: string; // ISO date string from Java LocalDate
  visited: boolean;
  hasPaidVisitFee?: boolean;
  visitFeeStatus?: "SI" | "NO" | "NO_SE_LE_COBRA" | "CANCELADA";
  visitFeeAmount?: number;
  paymentMethod?: string;
  invoiceNumber?: string;
  status: "SOLICITADA" | "SCHEDULED" | "VISITED" | "COMPLETED" | "CANCELLED";
  observations?: string;
}

File: crmuretek/src/main/java/com/crmuretek/crmuretek/models/Visit.java
package com.crmuretek.crmuretek.models;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;

@Entity
@Table(name = "leads_visits")
@Data
public class Visit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @ManyToOne
    @JoinColumn(name = "lead_id")
    private Consulta consulta;

    private LocalDate visitDate;
    private boolean hasPaidVisitFee;
    @Enumerated(EnumType.STRING)
    private VisitFeeStatus visitFeeStatus = VisitFeeStatus.NO;
    private Double visitFeeAmount;
    private String paymentMethod;
    private String invoiceNumber;

    @Enumerated(EnumType.STRING)
    private VisitStatus status = VisitStatus.SOLICITADA;

    @Column(columnDefinition = "TEXT")
    private String observations;
}
